hackathon_site/dashboard/frontend/src/slices/users/userSlice.ts

```diff
@@ -165,6 +216,34 @@ export const logout = createAsyncThunk<
     }
 });
 
+export const createProfile = createAsyncThunk<
+    Profile,
+    ProfileRequestBody,
+    { state: RootState; rejectValue: RejectValue; dispatch: AppDispatch }
+>(
+    `${userReducerName}/createProfile`,
+    async (profileRequestBody, { dispatch, rejectWithValue }) => {
+        try {
+            const response = await post<Profile>(
+                "/api/event/profiles/profile/",
+                profileRequestBody
+            );
+            return response.data;
+        } catch (e: any) {
+            dispatch(
+                displaySnackbar({
+                    message: `Failed to grant permissions to Hardware Signout Site: Error ${e.response.status}`,
```

Comment:
The message is kinda confusing from my view: something like "failed when granting permissions to *** to access HSS" would be a choice.

Reply:
changed!